EZSwiftExtensionsTests/ArrayTests.swift

```diff
@@ -285,6 +292,54 @@ class ArrayTests: XCTestCase {
         XCTAssertEqual(duplicatedElementsArrayOrderMaintained.unique(), [2, 1, 3])
     }
     
+    func testStep() {
+        XCTAssertNil(self.numberArray.step(-1))
+        XCTAssertNil(self.numberArray.step(0))
+
+        XCTAssertEqual(self.numberArray.step(2)!.count, 4)
+        XCTAssertEqual(self.numberArray.step(2)![0].count, 2) // count in each of the first child element will be <= stepSize
+        XCTAssertEqual(self.numberArray.step(2)![0], [0, 1])
+        XCTAssertEqual(self.numberArray.step(2)![1].count, 2) // count in each of the first child element will be <= stepSize
+        XCTAssertEqual(self.numberArray.step(2)![1], [2, 3])
+        XCTAssertEqual(self.numberArray.step(2)![2].count, 2) // count in each of the first child element will be <= stepSize
+        XCTAssertEqual(self.numberArray.step(2)![2], [4, 5])
+        XCTAssertEqual(self.numberArray.step(2)![3].count, 1)
+        XCTAssertEqual(self.numberArray.step(2)![3], [1])
+        
+        XCTAssertEqual(self.numberArray.step(3)!.count, 3)
+        XCTAssertEqual(self.numberArray.step(3)![0].count, 3) // count in each of the first child element will be <= stepSize
+        XCTAssertEqual(self.numberArray.step(3)![0], [0, 1, 2])
+        XCTAssertEqual(self.numberArray.step(3)![1].count, 3) // count in each of the first child element will be <= stepSize
+        XCTAssertEqual(self.numberArray.step(3)![1], [3, 4, 5])
+        XCTAssertEqual(self.numberArray.step(3)![2].count, 1)
+        XCTAssertEqual(self.numberArray.step(3)![2], [1])
+        
+        XCTAssertEqual(self.numberArray.step(7)!.count, 1)
+        XCTAssertEqual(self.numberArray.step(7)![0].count, self.numberArray.count) // count in each of the first child element will be <= stepSize
+        XCTAssertEqual(self.numberArray.step(7)![0], self.numberArray)
+        
+        XCTAssertEqual(self.numberArray.step(8)!.count, 1)
+        XCTAssertEqual(self.numberArray.step(8)![0].count, self.numberArray.count) // count in each of the first child element will be <= stepSize
+        XCTAssertEqual(self.numberArray.step(8)![0], self.numberArray)
+        
+        // multidimensional array [[0, 1], [2, 3], [4, 5], [1]]
```

Comment:
It would be better if the multi dimensional array tests live in a separate method. (Not a strong preference though)